### PR TITLE
Locked and Share indicators

### PR DESCRIPTION
**Pull request by ivan-nejezchleb**
JIRA: TNT-277
Locked and Share indicators in Dashboard's top bar.
Components can not be customized for now

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5928/ : SUCCESS in 29s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7709/ : SUCCESS in 16m 43s

File: libs/sdk-ui-dashboard/src/presentation/localization/bundles/en-US.json [hunk 1]
```diff
@@ -686,5 +686,30 @@
         "comment": "The user has a dashboard that contains no widgets.",
         "translate": false,
         "limit": 0
+    },
+    "header.lockStatus.tooltip": {
+        "value": "<b>This dashboard is locked.</b> Only administrators can edit this dashboard.",
+        "comment": "",
```
Comment (xMort) [inline libs/sdk-ui-dashboard/src/presentation/localization/bundles/en-US.json:697]: Comments for translators are missing.

**Comment by ivan-nejezchleb** [comment]
extended check cypress

**Comment by yenkins** [comment]
# ✨ Cypress test results ✨

## 🎉 Great news! Tests were successful.

|Status|Name|Tests|Failures|Time|
|:--:|:--|:--:|:--:|--:|
|✅|**drilling.spec.ts**|2|-|14.2670s|
|✅|**dashboard.spec.ts**|9|-|15.5060s|
|✅|**dateFilter.spec.ts**|7|-|21.6840s|

You can find full report [here](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/71/).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5947/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/71/ : SUCCESS in 12m 38s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5949/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7729/ : SUCCESS in 16m 47s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5951/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7733/ : SUCCESS in 16m 32s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5952/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7735/ : SUCCESS in 16m 36s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5956/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7739/ : FAILURE in 4m 49s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5959/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7742/ : SUCCESS in 18m 45s

File: libs/sdk-ui-dashboard/src/model/store/meta/metaSelectors.ts [hunk 2]
```diff
@@ -175,3 +179,16 @@ export const selectDashboardTags = createSelector(selectDashboardDescriptor, (st
...
+ *
+ * @alpha
+ */
+export const selectDashboardShareInfo = createSelector(selectDashboardDescriptor, (state) => {
+    const shareInfo: IAccessControlAware = {
+        shareStatus: state.shareStatus,
+        isLocked: state.isLocked,
+    };
+    return shareInfo;
+});
```
Comment (xMort) [inline libs/sdk-ui-dashboard/src/model/store/meta/metaSelectors.ts:194]: You can ignore this but you can save three lines by writing 

```
export const selectDashboardShareInfo = createSelector<AppState, IShareStatus, boolean, IAccessControlAware>(selectDashboardDescriptor, ({shareStatus, isLocked}) => ({
        shareStatus,
        isLocked,
}));
```

File: libs/sdk-ui-dashboard/src/presentation/topBar/shareIndicators/lockedStatus/DefaultLockedStatus.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import React from "react";
+
+import { ILockedStatusProps } from "./types";
+import { selectSettings, useDashboardSelector } from "../../../../model";
+import { LockedStatusIndicator } from "./LockedStatusIndicator";
+
+/**
+ * @alpha
+ */
+export const DefaultLockedStatus = (props: ILockedStatusProps): JSX.Element | null => {
```
Comment (xMort) [inline libs/sdk-ui-dashboard/src/presentation/topBar/shareIndicators/lockedStatus/DefaultLockedStatus.tsx:11]: The correct typing of functional component is `export const DefaultLockedStatus: React.FC<ILockedStatusProps> = (props) => {`

**Review by xMort — APPROVED** [review]

**Comment by xMort** [comment]
LGTM

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5963/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7746/ : FAILURE in 6m 39s

File: libs/sdk-ui-dashboard/src/model/store/meta/metaSelectors.ts [hunk 4]
```diff
@@ -184,3 +189,17 @@ export const selectDashboardShareStatus = createSelector(selectDashboardDescript
...
     return state.isLocked;
 });
+
+/**
+ * Selects complete dashboard share info.
+ *
+ * @alpha
+ */
+export const selectDashboardShareInfo = createSelector<
+    DashboardState,
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/model/store/meta/metaSelectors.ts:199]: Detail: do we need the explicit types?

File: libs/sdk-ui-dashboard/src/presentation/topBar/shareIndicators/lockedStatus/LockedStatusIndicator.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        return null;
+    }
+    return (
+        <div className="s-locked-status gd-locked-status">
+            <BubbleHoverTrigger>
+                <Icon.Lock
+                    className="gd-icon-locked"
+                    width={25}
+                    height={24}
+                    color={props.theme?.palette?.complementary?.c6 ?? "#94a1ad"}
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/presentation/topBar/shareIndicators/lockedStatus/LockedStatusIndicator.tsx:19]: Where does this default color come from? Or please convert it to a named constant to make eventual changes a bit more transparent.

File: libs/sdk-ui-dashboard/src/model/store/meta/metaSelectors.ts (hunk 4 above)
Comment (ivan-nejezchleb) [reply libs/sdk-ui-dashboard/src/model/store/meta/metaSelectors.ts:199]: return type needs to be defined explicitly, as I am picking particular values from whole DashboardDescriptor

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5967/ : SUCCESS in 15s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7750/ : FAILURE in 6m 43s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5971/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7754/ : FAILURE in 8m 03s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1965

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1718/ : FAILURE in 3m 30s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1999/ : SUCCESS in 41s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/796/ : SUCCESS in 14m 35s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7756/ : SUCCESS in 16m 26s

**Pull request merged**